### ISSUE TITLE
Maintenance Script

### ISSUE THREAD

**Issue by shawnrice** [shawnrice/alfred-bundler#62]
I'm opening this so that we can get some track of a few conversations that have popped up via email and things like #25.

Possibilities:
- Run the update script from the maintenance script
- Delete old copies of the bundler
# 

Agreed on functionality:
- Uninstall stale assets
- Verify icon / other image integrity
- Delete path cache entries for deleted assets

**Comment by shawnrice** [comment]
Just to reference the possibility from above. We could look to see if there are older versions of the bundler installed (for now, just `aries`). If so, we could consider walking through the installed workflow directories to see if one of the wrappers/bundlets for the old bundler exists. If we can't find any old wrappers/bundlets, we delete the old data and cache directories.

The processing there might be more intense in that the script would have to read through the wrapper/bundlet files to determine which version they are from. We could just select a few key lines that are unique and scan for those.

Thoughts?

**Comment by shawnrice** [comment]
Oh, and while we should definitely have a maintenance script in `Taurus 1`, we don't have to integrate all of the features yet as they could easily be implemented in `Taurus 2`, etc.... So, a full-feature wishlist completion isn't necessary here to push a release.

**Comment by shawnrice** [comment]
This script should also check the path cache and clear out entries when the path no longer exists (i.e. asset has been deleted).

**Comment by deanishe** [comment]
Detecting a lack of older versions of the bundler in workflows would be tricky: bundlets still work just fine if they're renamed.

I'd suggest creating a file somewhere in the bundler's datadir that can be "touched" every time a bundler is loaded. Makes it super easy to see when a bundler was last used without having to check the whole tree.

**Comment by shawnrice** [comment]
Good idea. It won't work for Aries, but it's good for going forward. Should we throw that in the data directory or the cache directory?

**Comment by deanishe** [comment]
Data. Where is won't get deleted.

Call it something obvious, like `last_loaded`.
